src/lib/variants.ts

```diff
@@ -161,8 +161,9 @@ export const userInTest = (mvtId: number): Filter => ({
     id: 'userInTest',
     test: (test, _) => {
         // Calculate range of mvtIDs for variants and return first match
-        const lowest = mvtId * (test.audienceOffset || 0);
-        const highest = lowest + mvtId * (test.audience || 1);
+        const maxMVTId = 1000000;
+        const lowest = maxMVTId * (test.audienceOffset || 0);
+        const highest = lowest + maxMVTId * (test.audience || 1);
```

Comment:
I definitely don't have a good understanding of how the multivariant logic works, so I'm almost certainly wrong here - but is the `highest` calculation here correct? I'd expect the highest to be at most `1000000`. Looking at the first test above we get `lowest` of `500,000` (i.e. 1,000,000 * 0.5). Then `highest` is `1,500,000` (i.e. 500,000 + (1,000,000 * 1). Do we need a `Math.min` in here somewhere?

Reply:
Yeah I wondered that too. The logic is pretty much copied exactly from Frontend's here: https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/experiments/ab-core.js#L54. It works fine, but it does result in values outside the possible range.

Reply:
:+1: Thanks for the explanation. Weird though - definitely not what I'd expect!

Reply:
Yeah, I think basically you're right that adding a Math.min would make the intent clearer, although it shouldn't change the behaviour as long as the function inputs are correct (are you don't pass in invalid MVT value).